test/functional/p2p_v2_transport.py

```diff
@@ -12,19 +12,116 @@
 
 class V2TransportTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.setup_clean_chain=True
-        self.num_nodes = 1
-        self.extra_args = [["-v2transport=0"]]
+        self.setup_clean_chain = True
+        self.num_nodes = 5
+        self.extra_args = [["-v2transport=1"], ["-v2transport=1"], ["-v2transport=0"], ["-v2transport=0"], ["-v2transport=0"]]
 
     def run_test(self):
-        network_info = self.nodes[0].getnetworkinfo()
+        sending_handshake = "start sending v2 handshake to peer"
+        downgrading_to_v1 = "retrying with v1 transport protocol for peer"
+        self.disconnect_nodes(0, 1)
+        self.disconnect_nodes(1, 2)
+        self.disconnect_nodes(2, 3)
+        self.disconnect_nodes(3, 4)
+
+        # verify local services
+        network_info = self.nodes[2].getnetworkinfo()
         assert_equal(int(network_info["localservices"], 16) & NODE_P2P_V2, 0)
         assert "P2P_V2" not in network_info["localservicesnames"]
-
-        self.restart_node(0, ["-v2transport=1"])
-        network_info = self.nodes[0].getnetworkinfo()
+        network_info = self.nodes[1].getnetworkinfo()
         assert_equal(int(network_info["localservices"], 16) & NODE_P2P_V2, NODE_P2P_V2)
         assert "P2P_V2" in network_info["localservicesnames"]
 
+        # V2 nodes can sync with V2 nodes
+        assert_equal(self.nodes[0].getblockcount(), 0)
+        assert_equal(self.nodes[1].getblockcount(), 0)
+        with self.nodes[0].assert_debug_log(expected_msgs=[sending_handshake],
+                                            unexpected_msgs=[downgrading_to_v1]):
+            self.connect_nodes(0, 1, peer_advertises_v2=True)
+        self.generate(self.nodes[0], 5, sync_fun=lambda: self.sync_all(self.nodes[0:2]))
+        assert_equal(self.nodes[1].getblockcount(), 5)
+        # verify there is a v2 connection between node 0 and 1
+        node_0_info = self.nodes[0].getpeerinfo()
+        node_1_info = self.nodes[0].getpeerinfo()
```

Comment:
@sipa, am I missing something or should this line read:

`node_1_info = self.nodes[1].getpeerinfo()`

Reply:
cc @theStack

Reply:
Good catch @kashifs. Do you want to open a PR for the fix? If not (for whatever reason), I'm happy to do it and add the commit with your authorship.

Reply:
I've opened a PR. Please let me know if it's done correctly and if there is more that I should do.